app/actions/service_credential_binding_create.rb

```diff
@@ -0,0 +1,68 @@
+module VCAP::CloudController
+  module V3
+    class ServiceCredentialBindingCreate
+      class UnprocessableCreate < StandardError
+      end
+
+      def initialize(user_audit_info)
+        @user_audit_info = user_audit_info
+      end
+
+      def precursor(service_instance, app: nil, name: nil)
+        not_supported! unless service_instance.user_provided_instance?
+        app_is_required! unless app.present?
+
```

Comment:
Not sure what type of validations go here instead of in the controller
e.g route does space mismatch check here rather than in the controller.

Reply:
I personally find weird that route binding is doing all of those validations in the action and not in the controller. Looking at other controllers (like service instances) I find most of the validation to be executed in the controller itself rather than in the action.

In this particular case, my reasoning here was to test the pre-requisites for the `precursor` method rather than the "validity" of the binding. That is, for this method to be executed, an app must be provided and the SI must be a UPSI.